### ISSUE TITLE
Simulation with non-dimensional units

### ISSUE THREAD

**Issue by navidcy** [CliMA/Oceananigans.jl#1462]
In the [Bickley jet example](https://clima.github.io/OceananigansDocumentation/dev/generated/Bickley_jet_shallow_water/), where we run the simulation we get the message:
```julia
[ Info: Simulation is stopping. Model time 2.500 minutes has hit or exceeded simulation stop time 2.500 minutes.
```
This is because we construct the simulation
```julia
Simulation(model, Δt = 1e-2, stop_time = 150)
```
it's assumed that 150 is 150 seconds. That could be a bit confusing. Is there is a way to tell the `Simulation` constructor that our units are nondimensional? Should we have such option?

**Comment by ali-ramadhan** [comment]
Yes this is indeed confusing. We probably do want a way of specifying the unit system of the simulation, also for NetCDF attributes. We discussed this a bit in #1362.

Seems like using Unitful.jl to define a unit for lengths (`grid.units`?) and a unit for time (`clock.units`?) might be enough?

I guess we might need a unit for mass, and a unit for each tracer as well?

But for this issue, a unit for time should be enough.

**Comment by glwagner** [comment]
Maybe `Simulation` needs a `metadata` property where we can put things like this.

**Comment by glwagner** [comment]
I think if we're going to keep track of units, we should store this in the grid (somehow)...

**Comment by glwagner** [comment]
Just like to discuss this a little more.

I think it'd be nice to have some awareness of units in the code. Some places we assume units, like `prettytime`. Other places we don't mention units, like `@show grid`.

The two important cases are SI units and "no units". Maybe we can just have `grid.units = u"m"` by default, where

```
julia> using Unitful

julia> typeof(u"m")
Unitful.FreeUnits{(m,), 𝐋, nothing
```

is the `Unitful.jl` way of saying "meters".

From the length of grid, we can infer the units for every other quantity. I think we should start unambitious and just have `grid.units` and `clock.units`; if `grid.units === u"m"` then we'll set `clock.units = u"s"`. We'll then use these units in things like `prettytime` and `@show grid`.

In principle, we also need `field.units`; field quantities have units that cannot be derived from `grid`. But for Oceananigans, units are really just sugar for showing and displaying things. So it may not make sense to add units to `field`.

**Comment by glwagner** [comment]
Thinking about it more I think we should have `grid.units = nothing` as default.

This is a downside for people who primarily do physical simulations. Because we're always printing correctly if we don't assume units (just verbosely, perhaps).

**Comment by glwagner** [comment]
I think we've decided not to do this, but @navidcy please re-openif you disagree!
